src/config.rs

```diff
@@ -314,6 +315,15 @@ fn construct_proxy_config(mc_path: &str, pc_env: Option<&str>) -> anyhow::Result
     Ok(pc)
 }
 
+pub fn empty_to_none<A: AsRef<str>>(inp: Option<A>) -> Option<A> {
```

Comment:
nit since it is Option<string>, can rewrite as 
pub fn empty_to_none (inp: Option<String>) -> Option<String> {